src/apps/the-exchange-app/components/ExchangeAction/ExchangeAction.tsx

```diff
@@ -0,0 +1,46 @@
+import TokenLogo from '../TokenLogo/TokenLogo';
+import Body from '../Typography/Body';
+import NumberText from '../Typography/NumberText';
+import ArrowRight from '../../images/arrow-right.png';
+import { useContext } from 'react';
+import { SwapDataContext } from '../../context/SwapDataProvider';
+import useGlobalTransactionsBatch from '../../../../hooks/useGlobalTransactionsBatch';
+
+const ExchangeAction = () => {
+    const { bestOffer, swapToken } = useContext(SwapDataContext);
+    const { addToBatch } = useGlobalTransactionsBatch();
+
+    const onClickToExchange = () => {
+        if (bestOffer) {
+            addToBatch({
+                title: 'Swap assets',
+                chainId: swapToken?.chainId || 0,
```

Comment:
Wondering what happens if chain is 0?!

Reply:
This needs to be better handled in the cleanup, that was a quick win